### PR TITLE
feat(ui): Create MaterialsCard Component (#61)

### LINKED ISSUE #61: Create MaterialsCard component
**Acceptance criteria:**
- Uses Chakra's `Box` component as the root element.
- Its props extend `BoxProps` from Chakra UI.
- Has 100% width by default.
- Receives the `status` prop with values: `idle`, `loading` and `generated`.

**Mockups:**
[Figma link](https://www.figma.com/file/ur0KLA9ZOtiodrAmpXrxh9/CodersCamp-app-and-website?node-id=634%3A1556)

### PR DESCRIPTION
**Pull request by talcia**
Closes #61

### CODE REVIEW

**Comment by KonradSzwarc** [comment]
When creating a PR remember to add "Closes #ISSUE_ID" to its description

File: packages/ui/src/components/MaterialsCard/MaterialsCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { MaterialsCard } from './MaterialsCard';
+
+describe('MaterialsCard', () => {
+  it('should renders correctly in idle status', () => {
+    render(<MaterialsCard />);
+
+    const button = screen.getByText('Wygeneruj listę');
+    expect(button).toBeEnabled();
+    expect(button).toBeInTheDocument();
+  });
```
Comment (KonradSzwarc) [inline packages/ui/src/components/MaterialsCard/MaterialsCard.test.tsx:13]: 🙃 For consistency, add new lines between tests

File: packages/ui/src/components/MaterialsCard/MaterialsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { Button } from '../Button';
+import { Typography } from '../Typography/Typography';
+
+type MaterialsCardStatus = 'idle' | 'loading' | 'generated';
+
+export interface MaterialsCardProps extends BoxProps {
+  status: MaterialsCardStatus;
+}
+
+type MaterialsCardOptions = { [status: string]: { statusText: string; buttonText: string } };
```
Comment (KonradSzwarc) [inline packages/ui/src/components/MaterialsCard/MaterialsCard.tsx:13]: 💭 You can type it as `Record<MaterialsCardStatus, { statusText: string; buttonText: string }>`

File: packages/ui/src/components/MaterialsCard/MaterialsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  generated: {
+    statusText: 'Twoja lista materiałów znajduje się na platformie Process.st',
+    buttonText: 'Przejdź do listy materiałów',
+  },
+};
+
+export const MaterialsCard = ({ status = 'idle', ...props }) => {
+  return (
+    <Box status={status} w="100%" boxShadow="base" borderRadius="8px" padding="24px" {...props}>
```
Comment (KonradSzwarc) [inline packages/ui/src/components/MaterialsCard/MaterialsCard.tsx:32]: 🤔 I'm not sure the Box component can receive a `status` prop. Why did you need to pass it here?

File: packages/ui/src/components/MaterialsCard/MaterialsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Box, BoxProps } from '@chakra-ui/react';
+
+import { Button } from '../Button';
+import { Typography } from '../Typography/Typography';
```
Comment (KonradSzwarc) [inline packages/ui/src/components/MaterialsCard/MaterialsCard.tsx:5]: 🙃 `import { Typography } from '../Typography';`

**Review by KonradSzwarc — CHANGES_REQUESTED** [review]

**Comment by talcia** [comment]
> @talcia When creating a PR remember to add "Closes #ISSUE_ID" to its description

Manually linking a pull request to an issue isn't enough?

File: packages/ui/src/components/MaterialsCard/MaterialsCard.test.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import { MaterialsCard } from './MaterialsCard';
+
+describe('MaterialsCard', () => {
+  it('should renders correctly in idle status', () => {
+    render(<MaterialsCard status="idle" />);
+
+    const button = screen.getByText('Wygeneruj listę');
```
Comment (hoolek77) [inline packages/ui/src/components/MaterialsCard/MaterialsCard.test.tsx:10]: 🤔  Shouldn't we import those texts? We've already defined them in the component file

**Review by hoolek77 — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/MaterialsCard/MaterialsCard.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+type MaterialsCardStatus = 'idle' | 'loading' | 'generated';
```
Comment (mrpodkalicki) [inline packages/ui/src/components/MaterialsCard/MaterialsCard.tsx:7]: 💭 I think better way is used ,,const enum'' because we used this value in a few spots (MaterialsCard, MateriaslCard.test, MaterialCard.stories) so if we'll want to change somtehing in future it will be easier. 
Second reason  is we don't have to  create again status object in MateriaCard.stories.tsx we'll just assign const enum. (line 7)

const MaterialsCardStatus = {
  idle: 'idle',
  loading: 'loading',
  generated: 'generated',
} as const;

**Review by hoolek77 — APPROVED** [review]

**Review by KonradSzwarc — APPROVED** [review]

**Comment by KonradSzwarc** [comment]
> @KonradSzwarc Manually linking a pull request to an issue isn't enough?

@talcia This way you link it only in ZenHub. When you write "Closes #ISSUE_ID" it's linked both in ZenHub task and GitHub issue. You can see the difference from the issue perspective below.
CleanShot 2021-07-28 at 16 36 51@2x

File: packages/ui/src/components/MaterialsCard/MaterialsCard.tsx (hunk 6 above)
Comment (KonradSzwarc) [reply packages/ui/src/components/MaterialsCard/MaterialsCard.tsx:7]: I'm not sure if it's beneficial as in such a case we would need to always have two imports (component and enum). The question is, how it makes future changes easier? If we change the name of one value we will need to replace it in every place it's used anyway.

Comment (mrpodkalicki) [reply packages/ui/src/components/MaterialsCard/MaterialsCard.tsx:7]: yes, but in my opinion, it is better to change the variable than a string and with this convention never miss a place that needs to be updated

Comment (KonradSzwarc) [reply packages/ui/src/components/MaterialsCard/MaterialsCard.tsx:7]: It seems like a really good argument when you create a JavaScript app where misspelling some string or forgetting to update it somewhere are quite common problems. I think when it comes to TypeScript those problems cannot happen as you will be notified in the console (and probably in your IDE as well).
